src/Lecture4.hs

```diff
@@ -196,8 +252,35 @@ row in the file.
    you can set the corresponding field to 'Nothing'.
 -}
 
+baseStats :: Row -> Stats
+baseStats r = Stats { statsTotalPositions = 1
+                    , statsTotalSum = Sum 0
+                    , statsAbsoluteMax = Max $ rowCost r
+                    , statsAbsoluteMin = Min $ rowCost r
+                    , statsSellMax = Nothing
+                    , statsSellMin = Nothing
+                    , statsBuyMax = Nothing
+                    , statsBuyMin = Nothing
+                    , statsLongest = MaxLen $ rowProduct r
+                    }
+
+withBuy :: Stats -> Row -> Stats
+withBuy s r = s { statsTotalSum = Sum $ rowCost r * (-1)
+                , statsBuyMax = Just . Max $ rowCost r
+                , statsBuyMin = Just . Min $ rowCost r
+                }
+
+withSell :: Stats -> Row -> Stats
+withSell s r = s { statsTotalSum = Sum $ rowCost r
+                 , statsSellMax = Just . Max $ rowCost r
+                 , statsSellMin = Just . Min $ rowCost r
+                 }
+
 rowToStats :: Row -> Stats
-rowToStats = error "TODO"
+rowToStats r
+  | rowTradeType r == Buy = withBuy (baseStats r) r
+  | rowTradeType r == Sell = withSell (baseStats r) r
+  | otherwise = Stats 0 0 (Max 0) (Min 0) Nothing Nothing Nothing Nothing (MaxLen "")
```

Comment:
Since `TradeType` can be only `Buy` or `Sell`, pattern matching works better here 🙂 

```suggestion
rowToStats r = case rowTradeType r of
    Buy  -> withBuy (baseStats r) r
    Sell -> withSell (baseStats r) r
```

It also will be better because if in the future you decide to add more constructors to the `TradeType`, GHC  will produce a compile-time warning to remind you to update this function. With the current implementation you won't get a warning.